lib/matplotlib/_type1font.py

```diff
@@ -768,13 +832,279 @@ def transform(self, effects):
+               (self._pos['Subrs'][0], subrs),
+               (self._pos['Encoding'][0], self._postscript_encoding(encoding))
+               ] + [(x, '') for x in self._pos.get('UniqueID', [])]
+        )
+        return type(self)((
+            newparts[0],
+            self._encrypt(newparts[1], 'eexec'),
             self.parts[2]
         ))
 
+    @staticmethod
+    def _charstring_tokens(data: T.Iterable[int]) -> T.Generator[int | str, None, None]:
+        """Parse a Type-1 charstring
+
+        Yield opcode names and integer parameters.
+        """
+        data = iter(data)
+        for byte in data:
+            if 32 <= byte <= 246:
+                yield byte - 139
+            elif 247 <= byte <= 250:
+                byte2 = next(data)
+                yield (byte-247) * 256 + byte2 + 108
+            elif 251 <= byte <= 254:
+                byte2 = next(data)
+                yield -(byte-251)*256 - byte2 - 108
+            elif byte == 255:
+                bs = bytes(itertools.islice(data, 4))
+                yield struct.unpack('>i', bs)[0]
+            elif byte == 12:
+                byte1 = next(data)
+                yield {
+                    0: 'dotsection',
+                    1: 'vstem3',
+                    2: 'hstem3',
+                    6: 'seac',
+                    7: 'sbw',
+                    12: 'div',
+                    16: 'callothersubr',
+                    17: 'pop',
+                    33: 'setcurrentpoint'
+                }[byte1]
+            else:
+                yield {
+                    1: 'hstem',
+                    3: 'vstem',
+                    4: 'vmoveto',
+                    5: 'rlineto',
+                    6: 'hlineto',
+                    7: 'vlineto',
+                    8: 'rrcurveto',
+                    9: 'closepath',
+                    10: 'callsubr',
+                    11: 'return',
+                    13: 'hsbw',
+                    14: 'endchar',
+                    21: 'rmoveto',
+                    22: 'hmoveto',
+                    30: 'vhcurveto',
+                    31: 'hvcurveto'
+                }[byte]
+
+    def _postscript_encoding(self, encoding: dict[int, str]) -> str:
+        """Return a PostScript encoding array for the encoding."""
+        return '\n'.join([
+            '/Encoding 256 array\n0 1 255 { 1 index exch /.notdef put} for',
+            *(
+                f'dup {i} /{glyph} put'
+                for i, glyph in sorted(encoding.items())
+                if glyph != '.notdef'
+            ),
+            'readonly def\n',
+        ])
+
+    def _subset_charstrings(self, glyphs: set[str]) -> str:
+        """Return a PostScript CharStrings array for the glyphs."""
+        charstrings = self.prop['CharStrings']
+        lenIV = self.prop.get('lenIV', 4)
+        encrypted = [
+            self._encrypt(charstrings[glyph], 'charstring', lenIV).decode('latin-1')
+            for glyph in glyphs
```

Comment:
Should this (and _subset_subrs below) sort the glyphs (and subrs) to ensure reproducibility?  (as set ordering changes over runs)

Reply:
The subrs are already in order (the loop is `for i in range(n_subrs)`) but sorting the glyphs is a good idea.